MAUI/DataForm/Editors.md

```diff
@@ -0,0 +1,1187 @@
+
+N> Class cannot be directly set as data type for autocomplete editor in this complex type scenario.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.ItemsSourceProvider = new DataFormItemsSourceProvider();
+    dataForm.DataObject = new ContactModel();
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+    dataForm.RegisterEditor("EmployeeDetails", DataFormEditorType.ComboBox);
+ 
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null && e.DataFormItem.FieldName == "EmployeeDetails" && e.DataFormItem is DataFormAutoCompleteItem autoCompleteItem)
+        {   
+            autoCompleteItem.DisplayMemberPath = "ID";
+            autoCompleteItem.SelectedValuePath = "Name";
+        }
+    } 
+ 
+    public class DataFormItemsSourceProvider  : IDataFormSourceProvider
+    {
+        public object GetSource(string sourceName)
+        {
+            if (sourceName == "EmployeeDetails")
+            {
+                List<EmployeeInfo> details = new List<EmployeeInfo>();
+                details.Add(new EmployeeInfo() { ID = 1, Name =  "Eric"  });
+                details.Add(new EmployeeInfo() { ID = 2, Name = "James"  });
+                details.Add(new EmployeeInfo() { ID = 3, Name = "Jacob"  });
+                details.Add(new EmployeeInfo() { ID = 4, Name = "Lucas"  });
+                details.Add(new EmployeeInfo() { ID = 5, Name = "Mark"  });
+                details.Add(new EmployeeInfo() { ID = 6, Name = "Aldan"  });
+                details.Add(new EmployeeInfo() { ID = 7, Name = "Aldrin"  });
+                details.Add(new EmployeeInfo() { ID = 8, Name = "Alan"  });
+                details.Add(new EmployeeInfo() { ID = 9, Name = "Aaron"  });
+
+                return details;
+            }
+        return new List<string>();
+        }
+    }
+
+    public class ContactInfo
+    {
+        [Display(Name ="First Name")]
+        public String FirstName { get; set; } 
+        public string EmployeeDetails { get; set; }
+    }
+
+    public class EmployeeInfo
+    {
+        public int ID { get; set; }
+        public string Name { get; set; }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+
+`Image`
+
+The `DataFormAutoCompleteEditor` provides display the filtered suggestions.
+
+The `AutoCompleteMode` property is used to choose the suggestion display mode in the `DataFormAutoCompleteItem` class.
+
+#### AutoComplete editor suggestion options
+
+The phenomenon of string comparison for filtering suggestions can be changed using the `"StartsWith"` and `"Contains"` property. The default filtering strategy is “StartsWith”, and it is case insensitive. The available filtering modes are,
```

Comment:
The string comparison for filtering suggestions can be changed using the `TextSearchMode ` property of `DataFormAutoCompleteItem`. The default text search strategy is “StartsWith”, and it is case insensitive. The available text search modes are,

* StartsWith
* Contains

Reply:
content updated